scs2-session-visualizer-jfx/src/main/java/us/ihmc/scs2/sessionVisualizer/jfx/controllers/yoPieChart/YoPieChartInputController.java

```diff
@@ -0,0 +1,302 @@
+   private Topic<List<String>> yoCompositeSelectedTopic;
+   private AtomicReference<List<String>> yoCompositeSelected;
+   private final String defaultText = "Drop YoVariable here";
+   private PieChart pieChart;
+
+   private YoVariableDatabase rootRegistryDatabase = null;
+
+   public void initialize(SessionVisualizerToolkit toolkit, Region rootPane, Labeled yoVariableDropLabel, Predicate<YoVariable> filter, PieChart pieChart)
+   {
+      this.rootPane = rootPane;
+      this.yoVariableDropLabel = yoVariableDropLabel;
+      this.filter = filter;
+      this.pieChart = pieChart;
+      this.yoCompositeSearchManager = toolkit.getYoCompositeSearchManager();
+      rootRegistryDatabase = toolkit.getYoManager().getRootRegistryDatabase();
+
+      messager = toolkit.getMessager();
+      yoCompositeSelectedTopic = toolkit.getTopics().getYoCompositeSelected();
+      yoCompositeSelected = messager.createInput(yoCompositeSelectedTopic);
+
+      // These property's get bound so it will update correctly
+      rootPane.styleProperty().bind(Bindings.concat(backgroundStyle, "; ", borderStyle));
+
+      rootPane.setOnDragDetected(this::handleDragDetected);
+      rootPane.setOnDragOver(this::handleDragOver);
+      rootPane.setOnDragDropped(this::handleDragDropped);
+      rootPane.setOnDragEntered(this::handleDragEntered);
+      rootPane.setOnDragExited(this::handleDragExited);
+      rootPane.setOnMousePressed(this::handleMousePressed);
+      rootPane.setOnMouseReleased(this::handleMouseReleased);
+   }
+
+   public void clear()
+   {
+      backgroundStyle.set(DEFAULT_BACKGROUND);
+      yoVariableDropLabel.setText(defaultText);
+   }
+
+   private void handleDragDetected(MouseEvent event)
+   {
+      if (event == null || yoVariable == null)
+         return;
+
+      if (!event.isPrimaryButtonDown())
+         return;
+
+      PickResult pickResult = event.getPickResult();
+
+      if (pickResult == null)
+         return;
+
+      Node intersectedNode = pickResult.getIntersectedNode();
+
+      if (intersectedNode == null)
+         return;
+
+      if (intersectedNode instanceof Text)
+      {
+         Text legend = (Text) intersectedNode;
+         String yoVariableName = legend.getText().split("\\s+")[0];
+         if (!yoVariableName.equals(yoVariableDropLabel.toString()))
+            return;
+         Dragboard dragBoard = legend.startDragAndDrop(TransferMode.COPY);
+         ClipboardContent clipboardContent = new ClipboardContent();
+         clipboardContent.put(DragAndDropTools.YO_COMPOSITE_REFERENCE, Arrays.asList(YoCompositeTools.YO_VARIABLE, yoVariable.getFullNameString()));
+         dragBoard.setContent(clipboardContent);
+      }
+
+      event.consume();
+   }
+
+   private void handleDragEntered(DragEvent event)
+   {
+      if (!event.isAccepted() && acceptDragEventForDrop(event))
+         setSelectionHighlight(true);
+      event.consume();
+   }
+
+   private void handleDragExited(DragEvent event)
```

Comment:
This allows for handling all the cases where the mouse enters and leaves the grid points